blog/_posts/2022-05-03-vscode-extension.md

```diff
@@ -0,0 +1,107 @@
+tags:
+- tiltfile
+- devtools
+- vscode
+---
+
+If you use [Visual Studio Code][code], are new to Tilt and have been spending a lot of time switching between writing your `Tiltfile` and reading our [`Tiltfile` API Reference][docs], stop! Go [install our new `Tiltfile` extension][install] and let us help you write `Tiltfile`s more easily! And if you're not a fan of VS Code, stay to learn about the components we used to build the extension. (You might be able to re-use what we built into Tilt to drive a `Tiltfile` extension in your own editor!)
+
+Our `Tiltfile` extension is [available in the Visual Studio marketplace][install], and can also be found in the Extensions sidebar inside VS Code. The extension requires you to have Tilt installed already; make sure you're using [the latest release][upgrade] to ensure you have all the features available.
+
+## Syntax Highlighting
+
+`Tiltfile`s are based on [Starlark][starlark], a dialect of Python. The extension automatically highlights Starlark thanks to a TextMate bundle derived from [Bazel's VS Code Extension][Bazel] and [Magic Python][].
+
+You can also use [our TextMate bundle for syntax highlighting with other IDEs that support it][tmbundle].
+
+## Completion and Signature Help
+
+A major reason for using an IDE like VS Code is to help you write code faster and more correctly with context-aware hints. The `Tiltfile` extension adds symbol completion for [Starlark builtin functions and methods][starlark], [Tilt builtins][docs], and any variables or functions you define in your `Tiltfile`s. The documentation shown in pop-ups is sourced from the same code that generates [our API docs][docs].
+
+When you type the opening parenthesis (`(`) of a function or method, the extension gives help on the parameters to that function:
+
+![Signature Help][imgsighelp]
+
+The extension has some basic type inference that will show only methods for Starlark strings, lists, or dictionaries when available, otherwise methods on all types will be presented.
+
+![Method Completion][imgmethods]
+
+## Hover and Go to Definition
+
+When you hover your cursor over a variable, function or method, the extension displays a documentation pop-up:
+
+![Hover Documentation][imghover]
+
+You can also type F12 ("Go to Definition") with the cursor on (almost) any symbol and jump to the definition of it. (Symbols for which you can't jump are Starlark/Tilt builtins and Tilt extensions. The latter can be fixed and [has an issue open that you can follow][refext].)
+
+## Load Statements
+
+The extension is aware of Starlark load statements in your `Tiltfile`s and makes the symbols you declare available for completion, signature help, and hover. It also can warn you about errors in your
+load statements:
+
+![Load Statements with errors][imgload]
+
+## Live Errors
+
+The extension detects when you have Tilt running, checks for any runtime errors in your Tiltfile, and underlines them:
```

Comment:
```suggestion
The extension detects when you have Tilt running, checks for any runtime errors in your Tiltfile, and underlines them. Since Tilt automatically re-runs your Tiltfile on change, this means you can edit your Tiltfile, hit save, and see errors right away, without leaving VS Code:
```

How about something like this to suggest the workflow?